CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+# Contributing guide
+
+We're always looking for new contributors and are happy to help you get started! Some ideas:
+
+- Add a test for a function.
+- Improve a function's documentation.
+- Work on any issue tagged [good first issue](https://github.com/remeda/remeda/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+- Work on any issue tagged [help wanted](https://github.com/remeda/remeda/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22).
+
+## Getting started
+
+Requirements:
+
+- Node.js `>=18.20.0`
+- npm `>=10.7.0`
+
+Install dependencies:
+
+```bash
+git clone git@github.com:remeda/remeda.git
+cd remeda
+npm install
+```
+
+Run a test, and watch for changes:
+
+```bash
+npm test -i src/file.test.ts
+```
+
+We have a pre-commit script that will format and lint the project.
+
+## Documentation
+
+Help us make the documentation better! See [`docs/README.md`](https://github.com/remeda/remeda/blob/main/docs/README.md).
+
+## Guidelines
+
+### Adding a new function
+
+When adding a new function, remember to:
+
+- Add a Typedoc comment with a description, parameters, signature, an example, and tags.
+- Add tests for runtime, typing, data-first, and data-last forms.
```

Comment:
Maybe add some more guidelines for testing, like preferring one assert per test, splitting typing and runtime tests (which you did mention). Using toEqualTypeOf and ts-expect-error for type tests?